shopfloor/services/zone_picking.py

```diff
@@ -383,48 +383,128 @@ def list_move_lines(self, zone_location_id, picking_type_id, order="priority"):
         return self._response_for_select_line(zone_location, picking_type, move_lines)
 
     def _scan_source_location(
-        self, zone_location, picking_type, location, order="priority"
+        self, zone_location, picking_type, barcode, order="prioriry"
     ):
-        """Return the move line related to the scanned `location`.
-
-        The method tries to identify unambiguously a move line in the location
-        if possible, otherwise return `False`.
+        """Search a location and find available lines into it.
         """
-        quants = self.env["stock.quant"].search([("location_id", "=", location.id)])
-        product = quants.product_id
-        lot = quants.lot_id
-        package = quants.package_id
+        response = None
+        message = None
+        search = self.actions_for("search")
+        location = search.location_from_scan(barcode)
+        if not location:
+            return response, message
+
+        if not location.is_sublocation_of(zone_location):
+            response = self._response_for_start(
+                message=self.msg_store.location_not_allowed()
+            )
+            return response, message
+
+        product, lot, package = self._find_product_in_location(location)
         if len(product) > 1 or len(lot) > 1 or len(package) > 1:
-            return False
+            response = self.list_move_lines(location.id, picking_type.id)
+            message = self.msg_store.several_products_in_location(location)
+            return response, message
+
         move_lines = self._find_location_move_lines(
-            location,
+            zone_location,
```

Comment:
I'm not sure to understand this change? If the user scan a location, it is to find the move lines within it, not the parent zone location?

Reply:
good catch! just a typo! :facepalm:

Reply:
fixed